src/collectionData.js

```diff
@@ -0,0 +1,109 @@
+export default class TodoListData {
+  constructor() {
+    this.todoListTasks = JSON.parse(window.localStorage.getItem('todoList')) || [];
+    this.taskRemoveButtons = [];
+  }
+
+  addTask(task) {
+    task.taskIndex = this.todoListTasks.length;
+    this.todoListTasks.push(task);
+    this.updateStorage();
+  }
+
+  removeTask(index) {
+    this.todoListTasks = this.todoListTasks.filter((task) => task.taskIndex !== index);
+    this.todoListTasks.forEach((task, i) => {
+      task.taskIndex = i;
+    });
+    this.updateStorage();
+  }
+
+  renderList(todoListWrapperElement, drag, drop, allowDrop) {
+    todoListWrapperElement.innerHTML = '';
+    this.todoListTasks.forEach((task, i) => {
+      task.taskIndex = i;
```

Comment:
- [ ] Kindly make sure always the `index` of the task starts at `1`. That means if we rendered the second task make sure its `index` is `2` instead of being `1`. Kindly implement that to meet the project requirement.

Reply:
fixed